fabrics.c

```diff
@@ -1122,21 +1125,18 @@ int nvmf_connect(const char *desc, int argc, char **argv)
 
 	nvme_parse_tls_args(keyring, tls_key, tls_key_identity, &cfg, c);
 
-	errno = 0;
 	ret = nvme_add_ctrl(h, c, &cfg);
-	if (ret)
+	if (ret) {
 		fprintf(stderr, "could not add new controller: %s\n",
-			nvme_strerror(errno));
-	else {
-		errno = 0;
-		if (flags != -EINVAL)
-			nvme_show_connect_msg(c, flags);
+			nvme_strerror(-ret));
+	} else if (flags != -EINVAL) {
```

Comment:
The existing flags checking not correct since never set `-EINVAL` to flags so always executed `nvme_show_connect_msg()` if `ret` set zero.

Reply:
Indeed, the flags check doesn't make sense. The flag can't be `EINVAL`, because we already filtered this out. It turns out that the device name is supposed to be printed on connect. I don't like it, IMO it should only printed with --verbose but it's too late to change this now. So lets keep it.

0ad49ade724e ("Print device name on connect")

Reply:
It turns out we can cleanup the error code path even more. Thanks for pointing it out!

Reply:
Thank you. The changes look good.